aiokafka/conn.py

```diff
@@ -262,8 +289,39 @@ def _do_sasl_handshake(self):
                     raise exc
                 auth_bytes = resp.sasl_auth_bytes
 
-        self.log.info(
-            'Authenticated as %s via PLAIN', self._sasl_plain_username)
+        if self._sasl_mechanism == 'GSSAPI':
+            self.log.info('Authenticated as %s via GSSPAI', self.principal)
+        else:
+            self.log.info('Authenticated as %s via PLAIN',
+                          self._sasl_plain_username)
+
+    @property
+    def principal(self):
+        service = self._sasl_kerberos_service_name
+        domain = self._sasl_kerberos_domain_name or self.host
+
+        return "{service}@{domain}".format(service=service, domain=domain)
+
+    def authenticator_gssapi(self):
+        name = gssapi.Name(self.principal,
+                           name_type=gssapi.NameType.hostbased_service)
+        cname = name.canonicalize(gssapi.MechType.kerberos)
+
+        client_ctx = gssapi.SecurityContext(name=cname, usage='initiate')
+
+        server_token = None
+        while not client_ctx.complete:
+            client_token = client_ctx.step(server_token)
+            if client_token:
+                server_token = yield client_token, True
+            else:
+                server_token = yield b'', True
+        msg = client_ctx.unwrap(server_token).message
+
+        iomsg = io.BytesIO(msg[0:1])
```

Comment:
Int8 is just 1 byte, right? msg[0] will give you the value already. This is full Python3, so no need for this wrapping.